vm/src/protocol/object.rs

```diff
@@ -514,10 +514,20 @@ impl PyObject {
     }
 
     pub fn hash(&self, vm: &VirtualMachine) -> PyResult<PyHash> {
+        // hash always exist
+        let hash = self.get_class_attr(identifier!(vm, __hash__)).unwrap();
```

Comment:
This line triggers python code and will increase runtime cost.
When hash is None, `cls.slots.hash` is pointing a default implementation raising error. Comparing pointers will be able to check the same thing.

Reply:
> This line triggers python code and will increase runtime cost. When hash is None, `cls.slots.hash` is pointing a default implementation raising error. Comparing pointers will be able to check the same thing.

May I ask how can I acquire the pointer pointing to the default error implementation?
I'm looking into the codes, but I have trouble finding the pointer.

Reply:
Oh, I found we cannot obtain function pointer of default implementation.

We don't seem to have a single pointer of `Unhashable::slot_hash`. Do you have a good idea?

Reply:
I think we need to check update_slot and hash_wrapper. If given one is None (singleton object), then use Unhashable::slot_hash (https://github.com/python/cpython/blob/main/Objects/typeobject.c#L8826-L8935)

Reply:
I am sorry, this comment was wrong.